src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.19;
+
+// External Interfaces
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {EnumerableSet} from "@oz/utils/structs/EnumerableSet.sol";
+
+// Internal Dependencies
+import {Module} from "src/modules/base/Module.sol";
+
+import {PaymentClient} from "src/modules/base/mixins/PaymentClient.sol";
+
+// Internal Interfaces
+import {IProposal} from "src/proposal/IProposal.sol";
+import {IRoleAuthorizer} from "src/modules/authorizer/IRoleAuthorizer.sol";
+import {IBountyManager} from "src/modules/logicModule/IBountyManager.sol";
+
+import {
+    IPaymentClient,
+    IPaymentProcessor
+} from "src/modules/base/mixins/PaymentClient.sol";
+
+// Internal Libraries
+import {LinkedIdList} from "src/common/LinkedIdList.sol";
+
+//@todo this has a direct dependency to the new RoleAuthorizer Module.
+//At the current state of the project the IRoleAuthorizer isnt the standardized Interface to use,
+//That means there is a possibility function calls will fail if the roleAuthorizer isnt implemented as the Authorizer
+//This will be changed in the coming future,
+//but till then the RoleAuthorizer has to be selected as the Authorizer Module of the proposal if the BountyManager is to be used
+
+contract BountyManager is IBountyManager, Module, PaymentClient {
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.UintSet;
+    using LinkedIdList for LinkedIdList.List;
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    //@todo Reminder that this will be moved into the Module Contract at a later point of time
+    modifier onlyRole(uint8 roleId) {
+        if (
+            !IRoleAuthorizer(address(__Module_proposal.authorizer()))
+                .isAuthorized(roleId, _msgSender())
+        ) {
+            revert Module__BountyManager__OnlyRole(roleId, address(this));
+        }
+        _;
+    }
+
+    modifier onlyClaimContributor(uint claimId) {
+        address sender = _msgSender();
+        Contributor[] memory contribs = _claimRegistry[claimId].contributors;
+        uint length = contribs.length;
+        bool found;
```

Comment:
You can save the cost of this extra boolean by comparing the for loop iterator with the length of the array after it gets out of the loop. Although in that case, do-while might be a more concrete choice.